File: android/src/main/java/com/jjlf/rngradient/SweepGradientView.java
package com.jjlf.rngradient;

import android.content.Context;
import android.view.ViewGroup;

public class SweepGradientView extends ViewGroup {

    private final GradientLayer mDrawable = new GradientLayer();
    SweepGradientView(Context context){
        super(context);
        setBackground(mDrawable);
        setClipChildren(false);
        mDrawable.setType(GradientLayer.SWEEP);
    }

    GradientLayer getDrawable() {
        return mDrawable;
    }
    @Override
    protected void onLayout(boolean changed, int l, int t, int r, int b) {

    }
}

File: android/src/main/java/com/jjlf/rngradient/SweepGradientView.kt
package com.jjlf.rngradient

import android.content.Context
import android.view.ViewGroup


class SweepGradientView(context: Context): ViewGroup(context) {


    private val mDrawable = GradientLayer()
    init{
        background = mDrawable
        clipChildren = false
        mDrawable.setType(GradientLayer.SWEEP)
    }

    fun getDrawable(): GradientLayer {
        return mDrawable
    }

    override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {

    }

}